openedx/core/lib/block_structure/transformer.py

```diff
@@ -125,4 +126,55 @@ def transform(self, usage_info, block_structure):
                 block structure, with already collected data for the
                 transformer, that is to be transformed in place.
         """
-        pass
+        raise NotImplementedError
+
+
+class OptimizedTransformer(BlockStructureTransformer):
```

Comment:
I'm not sure I like using NotImplementedError that way.  That implies that an abstract base class was not fully specified in the concrete class, and might cause linting issues.

I think I'd prefer to just implement that filter collection in the transform method.  Then the transformer manager just calls all the transform methods and runs a post-transform step to run all gathered filters.